pan3d/dataset_builder.py

```diff
@@ -386,12 +411,81 @@ def mesh_updated(exception=None):
     # Config logic
     # -----------------------------------------------------
 
-    def import_config(self, config_path):
-        with open(config_path) as config_file:
-            config = json.load(config_file)
-            self.set_dataset_path(dataset_path=config.get("dataset_path"))
-            self.state.update(config.get("state"))
+    def import_config(self, config_file):
+        if isinstance(config_file, str) and Path(config_file).exists():
+            with open(config_file) as f:
+                config = json.load(f)
+        else:
+            config = json.loads(config_file)
+        origin_config = config.get("data_origin")
+        array_config = config.get("data_array")
+        slices_config = config.get("data_slices")
+        ui_config = config.get("ui")
+
+        if not origin_config or not array_config:
+            self.state.dialog_message = "Invalid format of import file."
+            return
+
+        self.set_dataset_path(dataset_path=origin_config)
+        if "active" in array_config:
+            self.set_active_array(array_config["active"])
+        for axis in ["x", "y", "z", "t"]:
+            if axis in array_config:
+                self.state[axis + "_array"] = array_config[axis]
+        if "t_index" in array_config:
+            self.state.t_index = array_config["t_index"]
+
+        if slices_config:
+            coordinates = self.state.coordinates
+            for coordinate in coordinates:
+                if coordinate["name"] in slices_config:
+                    start, stop, step = slices_config[coordinate["name"]]
+                    coordinate["start"] = start
+                    coordinate["stop"] = stop
+                    coordinate["step"] = step
+            self.state.coordinates = coordinates
```

Comment:
remove `self.state.coordinates = coordinates`

Reply:
Removed in [ad725e2](https://github.com/Kitware/pan3d/pull/33/commits/ad725e2a63525dc5eb16b2779d48f51ec72d9cae)